diesel/src/query_dsl/mod.rs

```diff
@@ -287,6 +290,135 @@ pub trait QueryDsl: Sized {
         methods::SelectDsl::select(self, selection)
     }
 
+    /// Adds a `SELECT` clause to the query using based on a struct implementing [`Selectable`].
+    ///
+    /// If there was already a select clause present, it will be overridden.
+    /// For example, `foo.select(bar).select_by::<Baz>()` will produce the same
+    /// query as `foo.select_by::<Baz>()`, `foo.select_by::<Baz>().select(bar)`
+    /// as `foo.select(bar)` and `foo.select_by::<Bar>().select_by::<Baz>()` as
+    /// `foo.select_by::<Baz>()`
+    ///
+    /// `select_by` has slightly stricter bounds on its arguments than other
+    /// methods. In particular, the select clause constructed by `Selectable`
+    /// must be valid for the current query.
+    ///
+    /// [`Selectable`]: ../expression/trait.Selectable.html
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # include!("../doctest_setup.rs");
+    /// # use schema::users;
+    /// #
+    /// # #[derive(PartialEq, Eq, Debug, Clone, Queryable, Selectable)]
+    /// # #[table_name = "users"]
+    /// # pub struct UserName(#[column_name = "name"] String);
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use self::users::dsl::*;
+    /// #     let connection = establish_connection();
+    /// // By default, all columns will be selected
+    /// let all_users = users.load::<(i32, String)>(&connection)?;
+    /// assert_eq!(vec![(1, String::from("Sean")), (2, String::from("Tess"))], all_users);
+    ///
+    /// let all_names = users.select_by::<UserName>().load::<UserName>(&connection)?;
```

Comment:
Is it necessary to make this distinct between `select` and `select_by`? Is it possible for `select` to have this feature?

Reply:
@pickfire That's a PR, so that's where development happens. This means this is not the right place to ask questions. If you want to contribute something to this feature please make a fully fleshed out proposal how to implement that or at least how the API would look like, otherwise this is only unnecessary noise for all contributors.

Reply:
Yes, this is a PR. So can I know where should I ask question for the PR if I should not ask in the PR itself? Or am I not able to give a comment if I did a review or maybe I should not review a PR at all?

Reply:
Asking questions is about the PR itself is totally fine, but that's not what you've written above. You proposed something without a second tought (It cannot work that way, because how would you deal with having a variable amount of arguments to that function (one for the old style select and none for the new style one?)) and then you asked a completely unrelated  question about something different.
From comments on PR's I expect that they are about the PR itself and not about some unrelated problem you currently have. I'm happy to get proposals anywhere, as long as the person writing the proposal has spend some time to actually thing about how this would look like and how to implement that. (It can even be fine to just raise a problem like: "I've though about implementing it that way, but hit a problem there") Otherwise such things are just a huge waste of time for me.